packages/massa-web3/src/experimental/basicElements/operationManager.ts

```diff
@@ -149,35 +150,37 @@ export class OperationManager {
    */
   static deserialize(data: Uint8Array): OperationDetails {
     let offset = 0
-    const decodeNext = () => {
+    const nextVarInt = () => {
       const value = unsigned.decode(data, offset)
       offset += unsigned.encodingLength(value)
       return value
     }
 
     let operationDetails: BaseOperation = {
-      fee: decodeNext(),
-      expirePeriod: Number(decodeNext()),
-      type: Number(decodeNext()),
+      fee: nextVarInt(),
+      expirePeriod: Number(nextVarInt()),
+      type: Number(nextVarInt()),
     }
 
     switch (operationDetails.type) {
       case OperationType.Transaction: {
+        // decode address type to get encoded size
+        varint.decode(data, offset)
         const recipientAddress = Address.fromBytes(
-          data.slice(offset, offset + 33)
+          data.slice(offset, varint.decode.bytes + BLAKE3_HASH_BYTELENGTH)
         )
-        offset += 33
+        offset += varint.decode.bytes + BLAKE3_HASH_BYTELENGTH
```

Comment:
This logic is specific to the Address object and should be encapsulated there. Maybe using a `from transaction(data: Uint8Array, offset: number): Address, number`?

Reply:
i renamed the BLAKE3_HASH_BYTELENGTH to UNDERLYING_HASH_LEN and encapsulated the code to get the address lengh in getByteLength static method of Address